File: Classical/02-primeGenerator.cpp
#include <iostream>
#include <cmath>
using namespace std; 

bool isPrime(int n) 
{ 
    // Corner case 
    if (n <= 1){
        return false;
    }   
  
    // Check from 2 to n-1 
    for (int i=2; i< sqrt(n)+1; i+=2) {
        if (n%i == 0) {
            return false; 
        }
    }
    return true; 
} 

int primes(int a, int b){
    for (int i=a; i<=b; i++){
        if (isPrime(i)==1){
            return i;
        }
    }
}

int main(){
    int testcases, a, b;
    cin >> testcases;
    
    for (int j = 0; j < testcases; j++){
        cin >> a >> b;
        primes(a, b);
    }
}

File: Classical/02-primeGenerator.py
import numpy as np

def isPrime(n):
    if n>1:
        if n % 2 == 0 and n > 2: 
            return False
        return all(n % i for i in range(3, int(np.sqrt(n)) + 1, 2))
    else:
        return False

def primeTill(n):
    primes = []

    for i in range(n):
        if isPrime(i)==True:
            primes.append(i)

    return primes

if __name__ == "__main__":
    testcases = int(input())
    for i in range(testcases):
        
        a, b = input().split()
        a = int(a)
        b = int(b)

        for prime in primeTill(b):
            if prime>=a and prime<=b:
                print(prime)
